crates/noirc_abi/src/input_parser/json.rs

```diff
@@ -71,24 +81,43 @@ enum JsonTypes {
     Table(BTreeMap<String, JsonTypes>),
 }
 
-impl From<InputValue> for JsonTypes {
-    fn from(value: InputValue) -> Self {
-        match value {
-            InputValue::Field(f) => {
+impl JsonTypes {
+    fn try_from_input_value(
+        value: &InputValue,
+        abi_type: &AbiType,
+    ) -> Result<JsonTypes, InputParserError> {
+        let json_value = match (value, abi_type) {
+            (InputValue::Field(f), AbiType::Field | AbiType::Integer { .. }) => {
                 let f_str = format!("0x{}", f.to_hex());
                 JsonTypes::String(f_str)
             }
-            InputValue::Vec(v) => {
-                let array = v.iter().map(|i| format!("0x{}", i.to_hex())).collect();
-                JsonTypes::ArrayString(array)
-            }
-            InputValue::String(s) => JsonTypes::String(s),
-            InputValue::Struct(map) => {
-                let map_with_json_types =
-                    btree_map(map, |(key, value)| (key, JsonTypes::from(value)));
+            (InputValue::Field(f), AbiType::Boolean) => JsonTypes::Bool(f.is_one()),
```

Comment:
Looks like this may be the primary change. It seems like we're using the AbiType to supplement the InputValue kind because InputValue by itself doesn't contain enough information. Have we explored the alternate route of giving it this information by adding a `InputValue::Bool` instead?